geotools/src/main/scala/geotrellis/geotools/GridCoverage2DMultibandTile.scala

```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright (c) 2016 Azavea.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.geotools
+
+import geotrellis.raster._
+
+import org.geotools.coverage.grid._
+import org.geotools.gce.geotiff._
+import org.geotools.coverage.grid.io._
+
+
+/**
+  * The GridCoverage2DMultibandTile Object, houses a constructor.
+  */
+object GridCoverage2DMultibandTile {
+
+  /**
+    * Takes a GridCoverage2D and wraps it in a Geotrellis
+    * [[MultibandTile]]-derived object.
+    *
+    * @param  gridCoverage2D  The GeoTools GridCoverage2D object
+    */
+  def apply(gridCoverage: GridCoverage2D) =
+    new GridCoverage2DMultibandTile(gridCoverage)
+}
+
+/**
+  * Takes a GridCoverage2D and wraps it in a Geotrellis
+  * [[MultibandTile]]-derived object.
+  *
+  * @param  gridCoverage2D  The GeoTools GridCoverage2D object
+  */
+class GridCoverage2DMultibandTile(gridCoverage: GridCoverage2D)
+    extends MultibandTile with MacroMultibandCombiners {
+
+  val renderedImage = gridCoverage.getRenderedImage
+  val buffer = renderedImage.getData.getDataBuffer
+  val sampleModel = renderedImage.getSampleModel
+
+  private val _array = Array.ofDim[Int](sampleModel.getNumBands)
```

Comment:
Just a comment: This pretty much makes this not thread safe. We don't really write much multi-threaded code. Spark mostly sweeps things around so we have one-data per on-thread thingy. Not sure if `scala.util.DynamicVariable` would be justified here or even if `GridCoverage2D` is thread safe in the first place since they may fall back on the same pattern.